### PR TITLE
Allow adding GUIDs to each HSI security attr

### PR DESCRIPTION
**Pull request by hughsie**
This indicates the GUID in some way contributed to the result decided.

It also allows us to match the submitted HSI results back to a firmware
stream on the LVFS, which allows us to allow vendors to see a subset of
results for uploaded devices.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupd/fwupd-security-attr.c [hunk 1]
```diff
@@ -196,6 +197,91 @@ fwupd_security_attr_has_obsolete(FwupdSecurityAttr *self, const gchar *appstream
...
+ **/
+void
+fwupd_security_attr_add_guid(FwupdSecurityAttr *self, const gchar *guid)
+{
+	FwupdSecurityAttrPrivate *priv = GET_PRIVATE(self);
+	g_return_if_fail(FWUPD_IS_SECURITY_ATTR(self));
+	g_return_if_fail(guid != NULL);
+	if (fwupd_security_attr_has_guid(self, guid))
+		return;
+	g_ptr_array_add(priv->guids, g_strdup(guid));
```
Comment (superm1) [inline libfwupd/fwupd-security-attr.c:236]: what about `fwupd_guid_is_valid` here?

File: plugins/pci-bcr/fu-plugin-pci-bcr.c [hunk 2]
```diff
@@ -46,6 +55,8 @@ void
 fu_plugin_device_registered(FuPlugin *plugin, FuDevice *dev)
 {
 	FuPluginData *priv = fu_plugin_get_data(plugin);
+	if (fu_device_has_instance_id(dev, "main-system-firmware"))
+		g_set_object(&priv->msf_device, dev);
```
Comment (superm1) [inline plugins/pci-bcr/fu-plugin-pci-bcr.c:59]: should't this be put in the cache like it is for flashrom already?

Comment (superm1) [reply plugins/pci-bcr/fu-plugin-pci-bcr.c:59]: IOW change the below
		fu_plugin_cache_add(plugin, "main-system-firmware", dev);

File: src/fu-tool.c [hunk 3]
```diff
@@ -3041,6 +3050,14 @@ main(int argc, char *argv[])
...
 	       "exclude, e.g. 'internal,~needs-reboot'"),
 	     NULL},
+	    {"json",
+	     '\0',
+	     0,
+	     G_OPTION_ARG_NONE,
+	     &priv->as_json,
+	     /* TRANSLATORS: command line option */
+	     _("Output in JSON format"),
+	     NULL},
```
Comment (superm1) [inline src/fu-tool.c:3060]: this should probably split into a separate commit

File: libfwupd/fwupd-security-attr.c (hunk 1 above)
Comment (hughsie) [reply libfwupd/fwupd-security-attr.c:236]: return_val_if_fail perhaps?

File: plugins/pci-bcr/fu-plugin-pci-bcr.c (hunk 2 above)
Comment (hughsie) [reply plugins/pci-bcr/fu-plugin-pci-bcr.c:59]: Do you think the others should use fu_plugin_cache_add too?

Comment (superm1) [reply plugins/pci-bcr/fu-plugin-pci-bcr.c:59]: probably

File: libfwupd/fwupd-security-attr.c (hunk 1 above)
Comment (superm1) [reply libfwupd/fwupd-security-attr.c:236]: sure, that works

Comment (hughsie) [reply libfwupd/fwupd-security-attr.c:236]: Fixed.

File: src/fu-tool.c (hunk 3 above)
Comment (hughsie) [reply src/fu-tool.c:3060]: Fixed.

File: plugins/pci-bcr/fu-plugin-pci-bcr.c (hunk 2 above)
Comment (hughsie) [reply plugins/pci-bcr/fu-plugin-pci-bcr.c]: I converted both. I think the others make sense as strong refs.

**Review by superm1 — APPROVED** [review]
